plugins/org.komodo.shell/src/org/komodo/shell/commands/RenameCommand.java

```diff
@@ -44,89 +46,92 @@ public RenameCommand( final WorkspaceStatus wsStatus ) {
      */
     @Override
     protected CommandResult doExecute() {
-        String objNameArg=null;
-
         try {
-            objNameArg = requiredArgument( 0, I18n.bind( ShellI18n.invalidArgMsgObjectName ) );
-            String newName = requiredArgument( 1, I18n.bind( ShellI18n.invalidArgMsgNewName ) );
+            final String name = requiredArgument( 0, I18n.bind( ShellI18n.missingRenameFirstArg ) );
+            final String newChildName = optionalArgument( 1 ); // for renaming a child
+            final boolean renamingChild = !StringUtils.isBlank( newChildName );
+
+            KomodoObject objToRename = null;
+            String newName = null;
+            final KomodoObject context = getContext();
+
+            if ( renamingChild ) {
+                if ( !context.hasChild( getTransaction(), name ) ) {
+                    return new CommandResultImpl( I18n.bind( ShellI18n.childDoesNotExistToRename,
+                                                             context.getAbsolutePath(),
```

Comment:
recommend using   getWorkspaceStatus().getDisplayPath(context) here,
so the 'display path' is shown instead of real absolute path 'tko:komodo/tko:workspace/etc'